.github/workflows/ci_submodule.yml

```diff
@@ -0,0 +1,69 @@
+name: CI for libsc check
+on:
+  pull_request:
+
+jobs:
+
+  linux-sub:
+    permissions: write-all
+    if: |
+      github.event.pull_request.draft == false &&
+      github.event.pull_request.author_association != 'COLLABORATOR'
+    runs-on: ubuntu-latest
+    env:
+      TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      PR_NODE_ID: ${{ github.event.pull_request.node_id }}
+      INPUT_PATH: sc
+    steps:
+    - name: Checkout source code for base
+      uses: actions/checkout@v4
+      with:
+        submodules: false
+        ref: ${{ github.event.pull_request.base.ref }}
+        path: base
+
+    - name: Checkout source code for head
+      uses: actions/checkout@v4
+      with:
+        repository: ${{ github.event.pull_request.head.repo.full_name }}
+        ref: ${{ github.event.pull_request.head.ref }}
+        submodules: false
+        path: head
+        
+    - name: Determine hashes to compare
+      shell: bash
+      run: |
+        EVENT_PATH="${GITHUB_EVENT_PATH}"
+        cd "${GITHUB_WORKSPACE}" \
+        || error "__Line:${LINENO}__Error: Cannot change directory to Github Workspace"
+        PR=`jq -r ".number" "${EVENT_PATH}"`
+        cd head
+        HEAD_SC_HASH=`git log -n1 sc | awk 'NR==1{ print $2 }'`
+        cd ../base
+        BASE_SC_HASH=`git log -n1 sc | awk 'NR==1{ print $2 }'`
+        echo "HEAD_SC_HASH: $HEAD_SC_HASH, BASE_SC_HASH: $BASE_SC_HASH"
```

Comment:
Can we cd back into .. for a well defined situation after this line?

Reply:
Not after this, because we need to be in a repository to use github API functions later. But I fixed it before we exit the action for consistency.

Reply:
Ah cool. In that case we may not need the cd (it would have to be && not &, but let's leave it out).

Reply:
Alright, then I'll undo the changes.